tests/run/i22497.scala

```diff
@@ -0,0 +1,13 @@
+// scalajs: --skip
+
+import scala.annotation.publicInBinary
+import scala.annotation.experimental
+
+@experimental
+class Foo:
+  @publicInBinary private def this(i: Int) = this()
+  @publicInBinary protected def this(i: String) = this()
+
+@experimental
+@main def Test =
+  println(classOf[Foo].getConstructors().mkString("\n"))
```

Comment:
A more useful test would be to actually have an `inline` method that uses those constructors.